src/components/card/VideoCard.jsx

```diff
@@ -1,22 +1,81 @@
-import React from 'react';
+import React , { useState ,useEffect  }from 'react';
+import { useDispatch,useSelector } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
 import './VideoCard.css';
+import { MdOutlineWatchLater , MdPlaylistPlay } from 'react-icons/md'
+import { useRef } from 'react';
+import useClickOutside from '../../hooks/useClickOutside';
+import PlaylistModal from '../modal/PlaylistModal';
+import { openModal } from '../../features/modal/modalSlice';
+import { addSelectedVideo } from '../../features/playlist/playlistSlice';
+// const useClickOutside = (func) =>{
+// const domNode = useRef();
+//   useEffect(()=>{
+//       let handler = (e)=>{
+//         if(!domNode.current.contains(e.target) ) {
+//           func();
+//         }}
+//       document.addEventListener("mousedown",handler);
+//     return () =>{
+//       document.removeEventListener("mousedown",handler);
+//     }
+//   })
+//   return domNode;
+// } 
```

Comment:
Can shift this block of code to another utility if intending to use it in future or can remove it